### ISSUE TITLE
app_stopped_or_non_existent error when using app_direct in version 2.6.0

### ISSUE THREAD

**Issue by nicolasfguillaume** [openanalytics/shinyproxy#343]
Hello,

We deployed successfully our app to Shinyproxy 2.5.0, in app_direct mode, and everything works fine.

We uppgraded recently to Shinyproxy 2.6.0, still in app_direct mode, and now we are getting frequently this message when reloading the app :

```
{"status": "error", "message": "app_stopped_or_non_existent"}
```

PS: we don't have this error when the app run in "normal" mode (not app_direct mode).

Could you please assist?

Kind regards,

Nicolas

**Comment by LEDfan** [comment]
Hi

Thanks for opening this issue. This is indeed a new mechanism required for some of the new features in 2.6.0. The intention of course was that this does not break existing workflows. 
However, this should only happen when two conditions are met:
 - you are accessing a sub-page of the application (e..g. `http://localhost:8080/app_direct_i/flask-demo/_/about`)
 - the app does not send heartbeats. Usually an app sends heartbeast when it opens a websocket connection or when it regularly sends a HTTP request.

Is this correct for your situation as well? In that case, I'm afraid that the new behavior breaks your workflow. Currently there isn't a proper workaround, but you can try one of these things:
- do not use the `/app_direct` endpoint. Maybe you can explain a bit more why you use that endpoint instead of the `/app` endpoint. (note that you can hide the navbar on the `/app` endpoint as well).
- increase the `heartbeat-timeout` so that the app doesn't get automatically killed. I.e. when your app sends heartbeats every two minutes, increase the timeout to a bit longer than two minutes. However, if your app sends no heartbeats at all, this will not work.

I think that we can improve this system so that this does not break your workflow anymore, but this will require some development effort for the next release. I'll keep this issue open to track the progress of this improvement.

**Comment by gacsal** [comment]
Hi,

we are facing the same issue with v2.6.0.
And you are right, this occurs in this case:
"you are accessing a sub-page of the application"

We are happy if we can see a fix for this. Thank you!

**Comment by LEDfan** [comment]
Hi, today we released ShinyProxy 3.0.0. In this release the original behavior of the `app_direct` endpoint has been restored, such that works the same way as in 2.5.0. I hope this helps!
